### PR TITLE
Update version to 9.0.0

### PR DESCRIPTION
**Pull request by habbes**
Update version to 9.0.0 (official release).

### CODE REVIEW

File: src/Microsoft.AspNetCore.OData/Microsoft.AspNetCore.OData.csproj [hunk 1]
```diff
@@ -28,13 +28,13 @@
       <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
     </PackageReference>
     <PackageReference Include="Microsoft.OData.ModelBuilder" Version="1.0.9" />
-    <PackageReference Include="Microsoft.OData.Core" Version="8.0.0-rc.1" />
-    <PackageReference Include="Microsoft.OData.Edm" Version="8.0.0-rc.1" />
+    <PackageReference Include="Microsoft.OData.Core" Version="8.0.0" />
+    <PackageReference Include="Microsoft.OData.Edm" Version="8.0.0" />
+	  <PackageReference Include="Microsoft.Spatial" Version="8.0.0" />
```
Comment (gathogojr) [inline src/Microsoft.AspNetCore.OData/Microsoft.AspNetCore.OData.csproj:33]: nit: Alignment is off

File: test/Microsoft.AspNetCore.OData.E2E.Tests/Microsoft.AspNetCore.OData.E2E.Tests.csproj [hunk 2]
```diff
@@ -24,6 +24,9 @@
     <PackageReference Include="Moq" Version="4.13.1" />
     <PackageReference Include="xunit" Version="2.3.1" />
     <PackageReference Include="xunit.runner.visualstudio" Version="2.3.1" />
+    <PackageReference Include="Microsoft.OData.Core" Version="8.0.0" />
```
Comment (gathogojr) [inline test/Microsoft.AspNetCore.OData.E2E.Tests/Microsoft.AspNetCore.OData.E2E.Tests.csproj:27]: These references didn't exist before. Why are they needed now?

File: test/Microsoft.AspNetCore.OData.Tests/Microsoft.AspNetCore.OData.Tests.csproj [hunk 3]
```diff
@@ -24,6 +24,9 @@
     <PackageReference Include="xunit" Version="2.3.1" />
     <PackageReference Include="xunit.runner.visualstudio" Version="2.3.1" />
     <PackageReference Include="Microsoft.AspNetCore.TestHost" Version="8.0.6" />
+    <PackageReference Include="Microsoft.OData.Core" Version="8.0.0" />
+    <PackageReference Include="Microsoft.OData.Edm" Version="8.0.0" />
+    <PackageReference Include="Microsoft.Spatial" Version="8.0.0" />
```
Comment (gathogojr) [inline test/Microsoft.AspNetCore.OData.Tests/Microsoft.AspNetCore.OData.Tests.csproj:29]: These references didn't exist before. Why are they needed now?

File: test/Microsoft.AspNetCore.OData.E2E.Tests/Microsoft.AspNetCore.OData.E2E.Tests.csproj [hunk 4]
```diff
@@ -24,6 +24,9 @@
...
+    <PackageReference Include="Microsoft.OData.Edm" Version="8.0.0" />
+    <PackageReference Include="Microsoft.Spatial" Version="8.0.0" />
```
Comment (gathogojr) [inline test/Microsoft.AspNetCore.OData.E2E.Tests/Microsoft.AspNetCore.OData.E2E.Tests.csproj:29]: These references didn't exist before. Why are they needed now?

File: sample/ODataRoutingSample/ODataRoutingSample.csproj [hunk 5]
```diff
@@ -11,6 +11,7 @@
     <PackageReference Include="Microsoft.EntityFrameworkCore" Version="8.0.6" />
     <PackageReference Include="Microsoft.EntityFrameworkCore.InMemory" Version="8.0.6" />
     <PackageReference Include="Microsoft.EntityFrameworkCore.Proxies" Version="8.0.6" />
+    <PackageReference Include="Microsoft.OData.Edm" Version="8.0.0" />
```
[the remark this answers is not in the document]
Comment (xuzhg) [reply sample/ODataRoutingSample/ODataRoutingSample.csproj:14]: I think it's related to the version confliction. 
Microsoft.OpenApi.OData 1.0.6 depends ODL 7.x, meanwhile Microsoft.AspNetCore.OData project now depends on ODL 8.x. 
So, it seems the sample project needs this 'explicitly' package reference to know which one should be used in this dependency. 

I'd suggest to contact current owner of 'Microsoft.OpenApi.OData' to update the reference and have a new release on it.

File: test/Microsoft.AspNetCore.OData.E2E.Tests/Microsoft.AspNetCore.OData.E2E.Tests.csproj (hunk 4 above)
Comment (xuzhg) [reply test/Microsoft.AspNetCore.OData.E2E.Tests/Microsoft.AspNetCore.OData.E2E.Tests.csproj:29]: Same reason I think, the version confliction comes from "Microsoft.OData.ModelBuilder" and "Microsoft.AspNetCore.OData". 

That's why I was asking the similar question during yesterday's discussion

**Review by xuzhg — APPROVED** [review]

**Comment by xuzhg** [comment]
I am ok for the changes to resolve the warnings/errors. However, for an official release, we'd resolve the version confliction ahead.

**Comment by habbes** [comment]
I've created a PR to address the model builder conflicts: https://github.com/OData/ModelBuilder

For OpenAPI.OData conflict, we may need to sync with https://github.com/microsoft/OpenAPI.NET.OData

**Review by gathogojr — APPROVED** [review]

**Review by ElizabethOkerio — APPROVED** [review]

**Pull request merged**